### PR TITLE
feat: switch from scss to postcss

### PR DESCRIPTION
**Pull request by ovflowd**
## Description

To simplify the stack of the Node.js Website, I've decided to remove support for SASS/SCSS from this repository.

This is because all our styles were processed at least 3 times. (Sass -> PostCSS -> CSS -> Minification)

Introducing Tailwind would also create inconsistencies, as we use custom Tailwind "syntax" such as `@tailwind`, `@apply` and `theme`.

The SASS/SCSS processor also includes a lot of extra things that we don't necessarily need;

By switching to PostCSS, we wanted to retain a SCSS-alike syntax, such as supporting Mixins, Variables, Nesting, and more. This PR also adds relevant plugins **although** it is important to mention that the only significant difference is how `mixins` work with PostCSS.

The benefits of PostCSS include a lighter CSS footprint, faster compilation times, less verbose, and the support of a subset of functionalities (plugins) we like.

#### Differences in Mixins

In SCSS, you define a Mixin with `@mixin`, and if you have variables, you define them as `$variable`; With PostCSS, you define a Mixin with `@define-mixin`, and variables do not need the `$`.

In SCSS, you consume a Mixin with `@include`. In PostCSS, you consume a Mixin with `@mixin`.

### What changed in this PR?

- Full removal of SCSS/Sass support
  - Removed packages, renamed files from `*.scss` to `.css`
- Added PostCSS plugins with features that benefit our day-to-day development
- Updated any codebase that required to be patched for the functionality to work
- Removed a few unused SCSS files
- Updated Storybook `preview.tsx`, fixing styles, types, and other things
- Updated Storybook Dark/Light Mode support for the Preview Iframe
- Updated Storybook to use a default viewport
- Updated Docs and all references to SCSS to be PostCSS
- Updated relevant packages and package.json file
- Updated Stylelint Configuration
- Prettied/Linted all CSS files
- Added shareable minimal-compatible VSCode Configuration
- Updated Collaborator Guidelines with PostCSS
- Updated TypeScript configuration for Storybook

#### Why so many changes?

Some changes are partially unrelated to the migration from SCSS to PostCSS (mainly the Storybook changes); they're added in this PR primarily for convenience and to address these stylistic changes in a single batch. 

## Validation

Storybooks, Builds, and Local Development should work as expected.

### CODE REVIEW

**Review by AugustinMauroy — APPROVED** [review]
Sounds good to me 👍

File: COLLABORATOR_GUIDE.md [hunk 1]
```diff
@@ -172,7 +177,7 @@ Finally, if you're unfamiliar with how to use Tailwind or how to use Tailwind wi
 
 > [!NOTE]\
 > Tailwind is already configured for this repository. You don't need to import any Tailwind module within your CSS modules.
-> You can apply Tailwind classes by regularly using CSS variables or SCSS's `@apply` statement, for example. If you have questions, please raise them on Pull Requests or Issues.
+> You can apply Tailwind classes by regularly using CSS variables or PostCSS's `@apply` statement, for example. If you have questions, please raise them on Pull Requests or Issues.
```
Comment (ovflowd) [inline COLLABORATOR_GUIDE.md]:
```suggestion
> You can apply Tailwind classes by regularly using CSS variables or Tailwind's `@apply` token, for example. If you have questions, please raise them on Pull Requests or Issues.
```

File: package.json [hunk 2]
```diff
@@ -49,6 +49,7 @@
     "autoprefixer": "~10.4.15",
     "classnames": "~2.3.2",
     "cross-env": "7.0.3",
+    "cssnano": "^6.0.1",
```
Comment (ovflowd) [inline package.json]: Oh, this should be changed to `6.0.1` instead of `^6.0.1`

File: postcss.config.js [hunk 3]
```diff
@@ -1,6 +1,11 @@
 module.exports = {
   plugins: {
+    'postcss-mixins': {},
+    'postcss-simple-vars': {},
+    'postcss-import': {},
+    'tailwindcss/nesting': {},
     tailwindcss: {},
     autoprefixer: {},
+    cssnano: {},
```
Comment (ovflowd) [inline postcss.config.js:10]: We might in the future consider applying this plugin only if ENV is `production` if our CSS becomes too big, due to performance implications.

**Review by shanpriyan — APPROVED** [review]
Looks good to me 👍🏽🚀

**Comment by ovflowd** [comment]
cc @nodejs/web-infra I'd also like approval from the Web Infra team as this is sorta of a infrastructure change.

**Comment by AugustinMauroy** [comment]
# Preview:
Capture d’écran 2023-09-24 à 10 50 25

# actual deploy:

Capture d’écran 2023-09-24 à 10 50 47

You should take a look why layout had change. _it's just on index page_

**Comment by ovflowd** [comment]
> # Preview:
> Capture d’écran 2023-09-24 à 10 50 25
> # actual deploy:
> Capture d’écran 2023-09-24 à 10 50 47
> You should take a look why layout had change. _it's just on index page_

I think I've found the issue.

**Comment by AugustinMauroy** [comment]
@ovflowd FIX on lasted commit 👍

**Comment by ovflowd** [comment]
cc @nodejs/nodejs-website I've decided to commit (include) basic shared VSCode settings (by committing `.vscode`) I wanted your opinions on this matter, as by doing this, our repository would include IDE-specific settings within the repository.

Please let me know what you all think of this.

File: .vscode/extensions.json [hunk 4]
```diff
@@ -0,0 +1,9 @@
+{
+  "recommendations": [
+    "esbenp.prettier-vscode",
+    "bradlc.vscode-tailwindcss",
+    "stylelint.vscode-stylelint",
+    "unifiedjs.vscode-mdx",
+    "dbaeumer.vscode-eslint"
```
Comment (bmuenzenmeyer) [inline .vscode/extensions.json]: We also use editorconfig

```suggestion
    "dbaeumer.vscode-eslint",
    "editorconfig.editorconfig"
```

File: .vscode/settings.json [hunk 5]
```diff
@@ -0,0 +1,9 @@
+{
+  "css.validate": false,
+  "editor.formatOnSave": true,
+  "editor.defaultFormatter": "esbenp.prettier-vscode",
+  "files.insertFinalNewline": true,
+  "files.trimTrailingWhitespace": true,
```
Comment (bmuenzenmeyer) [inline .vscode/settings.json]: these seem to overlap with editorconfig settings, yes?

https://github.com/nodejs/nodejs.org/blob/main/.editorconfig#L8-L9

File: COLLABORATOR_GUIDE.md [hunk 6]
```diff
@@ -140,45 +145,57 @@ React allows us to create user interfaces with a modern take on Web Development.
 
 If you're unfamiliar with React or Web Development in general, we encourage a read before taking on complex issues and tasks as this repository is **not for educational purposes** and we expect you to have a basic understanding of the technologies used.
 
-We also recommend getting familiar with technologies such as [Next.js][], [MDX][], [SCSS][], and "concepts" such as "CSS Modules" and "CSS-in-JS".
+We also recommend getting familiar with technologies such as [Next.js][], [MDX][], [PostCSS][], and "concepts" such as "CSS Modules" and "CSS-in-JS".
 
 ### Styling a Component
 
-As mentioned, we write all Component-styles in a separate `.module.scss` files. This is like writing any CSS in a separate file (besides the fact that we use [SCSS][]).
+As mentioned, we write all Component-styles in a separate `.module.css` files. This is like writing any CSS in a separate file (besides the fact that we use [PostCSS][]).
```
Comment (bmuenzenmeyer) [inline COLLABORATOR_GUIDE.md]: grammar
```suggestion
As mentioned, we write all Component-styles in separate `.module.css` files. This is like writing any CSS in a separate file (besides the fact that we use [PostCSS][]).
```

File: COLLABORATOR_GUIDE.md [hunk 7]
```diff
@@ -140,45 +145,57 @@ React allows us to create user interfaces with a modern take on Web Development.
...
+> \[!IMPORTANT]\
 > For styling Components, only using Tailwind variables for defining paddings, margins, spacing, flexboxes, and all sorts of CSS classes is mandatory.
 > There are times when manual values are allowed, but we recommend setting them in reusable variables. Within the `styles` folder.
 > If unsure how to proceed, ask for help within Issues or Pull Requests.
 
+> \[!IMPORTANT]\
+> When using IDEs such as Visual Studio Code, we recommend installing the official [Stylelint](https://marketplace.visualstudio.com/items?itemName=bradlc.vscode-tailwindcss)
+> and [Tailwind](https://marketplace.visualstudio.com/items?itemName=stylelint.vscode-stylelint) Extensions.\
+> This is necessary since we use custom CSS syntax (that comes from our PostCSS plugins)\
+> Our Stylelint Configuration is already compatible with our PostCSS rules and plugins
```
Comment (bmuenzenmeyer) [inline COLLABORATOR_GUIDE.md]: grammar

```suggestion
> Our Stylelint Configuration is already compatible with our PostCSS rules and plugins.
```

File: COLLABORATOR_GUIDE.md [hunk 8]
```diff
@@ -367,6 +384,33 @@ They are React Components that receive the `children` prop, which is the transfo
...
+- `postcss-nested`: Allows us to use Sass-like Nesting
+- `postcss-calc`: Strips `calc` expressions and replaces them with the result
+
+It is important to mention that even though we use SCSS-like syntax, we do not use SCSS, and some of these plugins
+are not 100% compatible with the SCSS syntax.
+For example, `postcss-mixins` does not support `@include` and `@extend` directives (and it uses `@define-mixin` for defining Mixins and `@mixin` for including Mixins).
+
+#### What other benefits PostCSS brings?
+
+- Our current CSS Framework, [Tailwind][] uses PostCSS for its transpilation process
```
Comment (bmuenzenmeyer) [inline COLLABORATOR_GUIDE.md]: grammar

```suggestion
- Our current CSS Framework, [Tailwind][], uses PostCSS for its transpilation process
```

File: COLLABORATOR_GUIDE.md [hunk 9]
```diff
@@ -388,13 +432,13 @@ This means that Shiki's Themes and Languages are not bundled by default.
...
-### Tailwind
+### TailwindCSS
 
-Tailwind is a great CSS Framework. It allows us to create a Design System that is easy to maintain and extend. It also allows us to create a consistent Design Language across the Website.
+[Tailwind][] is a great CSS Framework. It allows us to create a Design System that is easy to maintain and extend. It also allows us to create a consistent Design Language across the Website.
 
 For example, it automatically integrates with PostCSS and Next.js, which allows us to use Tailwind's utilities within our Components and Pages.
 
-It also integrates with Sass allowing to use Tailwind classes directly within our SCSS Modules.
+It also integrates with Sass allowing to use Tailwind classes directly within our CSS Modules.
```
Comment (bmuenzenmeyer) [inline COLLABORATOR_GUIDE.md]: Sass reference that might not make sense. please review

**Review by bmuenzenmeyer — APPROVED** [review]
some small nits worth changing

File: .vscode/extensions.json (hunk 4 above)
Comment (ovflowd) [reply .vscode/extensions.json]: Prettier automatically loads EditorConfig (afaik) and VSCode also automatically loads editorconfig, no?

File: .vscode/settings.json (hunk 5 above)
Comment (ovflowd) [reply .vscode/settings.json]: Feel free to remove them (I can also do it)

File: COLLABORATOR_GUIDE.md (hunk 9 above)
Comment (ovflowd) [reply COLLABORATOR_GUIDE.md]: Feel free to remove!

File: package.json [hunk 10]
```diff
@@ -58,6 +59,10 @@
     "next-mdx-remote": "^4.4.1",
     "next-themes": "^0.2.1",
     "postcss": "~8.4.30",
+    "postcss-calc": "~9.0.1",
+    "postcss-import": "~15.1.0",
+    "postcss-mixins": "~9.0.4",
+    "postcss-simple-vars": "~7.0.1",
```
Comment (MattIPv4) [inline package.json:65]: My only thought here would be if this is needed, or if native CSS vars could be used?

Comment (ovflowd) [reply package.json:65]: CSS variables work differently from SCSS variables and are limited/constrained. This allows us to have a Sassa-like functionality.

Comment (MattIPv4) [reply package.json:65]: Ah, okay 👍

**Review by MattIPv4 — APPROVED** [review]
Seems like a logical change to me, I'm always a fan of less tools in the chain

**Comment by ovflowd** [comment]
@bmuenzenmeyer and @MattIPv4 if you two don't mind, could y'all give a last copy-check on the changes? I've just pushed some changes to simplify and better some docs

**Comment by bmuenzenmeyer** [comment]
> if you two don't mind, could y'all give a last copy-check on the changes?

✅ looking good to me

**Review by Harkunwar — APPROVED** [review]

**Pull request merged**